pkg/build/controller/common/util.go

```diff
@@ -134,3 +140,61 @@ func HasBuildPodNameAnnotation(build *buildapi.Build) bool {
 	_, hasAnnotation := build.Annotations[buildapi.BuildPodNameAnnotation]
 	return hasAnnotation
 }
+
+// ProcessValueFrom supports the k8s pod syntax of allowing "valueFrom" in addition
+// to "value", where "valueFrom" uses the downward api to reference existing env
+// variables, jsonpath references to the build object, secrets, and configmaps.
+func ProcessValueFrom(build *buildapi.Build, client kclientset.Interface) (*[]kapi.EnvVar, error) {
+	var outputEnv []kapi.EnvVar
+	var allErrs []error
+
+	inputEnv := buildutil.GetEnvFromStrategy(build)
+	store := envutil.NewResourceStore()
+
+	for _, e := range *inputEnv {
+		var value string
+		var err error
+
+		if e.Value != "" {
+			r, err := regexp.Compile(`(\$\([A-Z_]+\))`)
```

Comment:
This section is quite problematic.  Please take a look at the use of `k8s.io/kubernetes/third_party/forked/golang/expansion.Expand()` in vendor/k8s.io/kubernetes/pkg/kubelet/kubelet_pods.go and use that as a model.